internal/dag/builder.go

```diff
@@ -514,6 +514,16 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 		secure := b.lookupSecureVirtualHost(host)
 		addRoutes(secure, routes)
 	}
+
+	// HTTPProxy object must have either a TCPProxy spec, or
```

Comment:
Can this check move higher, around 507?

Reply:
Yeh that makes sense if `computeRoutes` returns an error.